tests/phpunit/Process/Runner/TestRunConstraintCheckerTest.php

```diff
@@ -202,4 +202,85 @@ public function test_run_passes_on_exactly_covered_min_msi(): void
 
         $this->assertTrue($constraintChecker->hasTestRunPassedConstraints());
     }
+
+    /** @dataProvider is_msi_over_minimum_msi_provider */
+    public function test_is_msi_over_minimum_msi(?float $actualMsi, ?float $actualCoveredMsi, float $minMsi, float $minCoveredMsi, string $type, bool $expected): void
+    {
+        $metrics = $this->createMock(MetricsCalculator::class);
+
+        if ($actualMsi !== null) {
+            $metrics->expects($this->once())->method('getMutationScoreIndicator')->willReturn($actualMsi);
+        }
+
+        if ($actualCoveredMsi !== null) {
+            $metrics->expects($this->once())->method('getCoveredCodeMutationScoreIndicator')->willReturn($actualCoveredMsi);
+        }
+
+        $constraintChecker = new TestRunConstraintChecker(
+            $metrics,
+            false,
+            $minMsi,
+            $minCoveredMsi
+        );
+
+        $this->assertSame($expected, $constraintChecker->isMsiOverMinimumMsi($type));
+    }
+
+    public function is_msi_over_minimum_msi_provider(): array
```

Comment:
We generally use a Generator for data providers.

Data provider also aren't snake cased, but use normal camel case